Settings

lagbox opened the pull request:
A persistent settings system.

lagbox committed:
Minor adjustments

lagbox commented:
I will squash these down when ready.

I need to test this middleware still.

lagbox commented on app/Data/database/migrations/2016_02_12_070250_create_settings_table_migration.php:
```diff
@@ -0,0 +1,34 @@
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Database\Migrations\Migration;
+
+class CreateSettingsTableMigration extends Migration
```
@Nitpick-CI well thats like your opinion man.

ryanwinchester replied:
haha

ryanwinchester commented:
Is this good to go?

ryanwinchester commented:
Also, I came across this:

https://laravel.com/api/5.1/Illuminate/Foundation/Http/Kernel.html#method_pushMiddleware

```php
public function boot(\Illuminate\Contracts\Http\Kernel $kernel)
{
     $kernel->appendMiddleware('Sheepy85\L5Localization\Middleware\Localization');
}
```

as a way to register the middleware in the ServiceProvider :)

ryanwinchester commented:
I came across this:

https://laravel.com/api/5.1/Illuminate/Foundation/Http/Kernel.html#method_pushMiddleware

``` php
public function boot(\Illuminate\Contracts\Http\Kernel $kernel)
{
     $kernel->pushMiddleware(\Flashtag\Data\Settings\SettingsMiddleware::class);
}
```

as a way to register the middleware in the ServiceProvider :)

lagbox commented:
Pushing the middleware works just fine. Tested the middleware as well.

lagbox committed:
Settings - getting dirty

oh so dirty, yet oh so clean, and we can prove it

Note: adjusted sqlite database path to use database_path
